api/v01/vrp.rb

```diff
@@ -170,48 +171,42 @@ class Vrp < APIBase
             id = params[:id]
             job = Resque::Plugins::Status::Hash.get(id)
             stored_result = APIBase.dump_vrp_dir.read([id, params[:api_key], 'solution'].join('_'))
-            solution = stored_result && Oj.load(stored_result) rescue Marshal.load(stored_result) # rubocop:disable Security/MarshalLoad, Style/RescueModifier
+            job_object = stored_result && Oj.load(stored_result) rescue Marshal.load(stored_result) # rubocop:disable Security/MarshalLoad, Style/RescueModifier
 
-            if solution.nil? && (job.nil? || job.killed? || Resque::Plugins::Status::Hash.should_kill?(id) || job['options']['api_key'] != params[:api_key])
+            if job_object.nil? && (job.nil? || job.killed? || Resque::Plugins::Status::Hash.should_kill?(id) || job['options']['api_key'] != params[:api_key])
               status 404
               error!({ message: "Job with id='#{id}' not found" }, 404)
             end
 
-            solution ||= OptimizerWrapper::Result.get(id) || {}
+            job_object ||= OptimizerWrapper::Result.get(id) || {}
             output_format = params[:format]&.to_sym ||
-                            (solution[:configuration] && solution[:configuration][:csv] ? :csv : env['api.format'])
+                            (job_object[:configuration] && job_object[:configuration][:csv] ? :csv : env['api.format'])
             env['api.format'] = output_format # To override json default format
 
             if job&.completed? # job can still be nil if we have the solution from the dump
-              APIBase.dump_vrp_dir.write([id, params[:api_key], 'solution'].join('_'), Oj.dump(solution)) if stored_result.nil? && OptimizerWrapper.config[:dump][:solution]
+              APIBase.dump_vrp_dir.write([id, params[:api_key], 'solution'].join('_'), Oj.dump(job_object)) if stored_result.nil? && OptimizerWrapper.config[:dump][:solution]
               OptimizerWrapper.job_remove(params[:api_key], id)
             end
 
             status 200
 
-            unless solution.nil? || solution[:result].nil? || solution[:result].is_a?(Array)
-            # TODO: solution[:result] should always be an array, find out why it is not and
-            # remove this check and other similar ones -- i.e.,  [solution[:result]].flatten(1).
-              solution[:result] = [solution[:result]]
-            end
-
             if output_format == :csv && (job.nil? || job.completed?) # At this step, if the job is nil then it has already been retrieved into the result store
-              present(OutputHelper::Result.build_csv(solution[:result]), type: CSV)
+              present(OutputHelper::Result.build_csv(job_object[:result].map(&:deep_symbolize_keys!)), type: CSV)
```

Comment:
`deep_symbolize_keys` is not free, we can load the object with symbolised key directly, Oj.load shoudl have an option for that.

Reply:
actually if we can do `job_object[:result]` why the rest is not already symbolized :thinking: ?